www/js/main.js

```diff
@@ -37,75 +34,117 @@ define([
 		}
 	
-	//The following code follows sachin code after the model has been rendered according to query parameters
-	ready(function(){
-		//remove the loading division, now that the problem is being loaded
-		var loading = document.getElementById('loadingOverlay');
-		loading.style.display = "none";
-
-		var dm = new drawModel(_model.active);
-
-		var menuButtons=[];
-		menuButtons.push("createQuantityNodeButton");
-		menuButtons.push("createEquationNodeButton");
-
-		array.forEach(menuButtons, function(button){
-			//setting display for each menu button
-			style.set(registry.byId(button).domNode, "display", "inline");
-
-			/*
-		 	* This is a work-around for getting a button to work inside a MenuBar.
-		 	* Otherwise, there is a superfluous error message.
-		 	*/	
-		 	registry.byId(button)._setSelected = function(arg){
-				console.log(button+" _setSelected called with ", arg);
-			}
+		ready(function(){
+			//remove the loading division, now that the problem is being loaded
+			var loading = document.getElementById('loadingOverlay');
+			loading.style.display = "none";
+
+			var dm = new drawModel(_model.active);
+
+			/*********  below this part are the event handling *********/
+			aspect.after(dm, "onClickNoMove", function(mover){
+				if(mover.mouseButton != 2){
+					// show node editor only when it is not a right click
+					// TODO: attach node editor click
+					console.log("node open action for ", mover.node.id);
+				} else {
+					// TODO: attach menu display click event
+					console.log("menu open action for ", mover.node.id);
+				}
+			}, true);
+
+			aspect.after(dm, "onClickMoved", function(mover){
+				console.log("aspect after called ", mover);
+				var g = geometry.position(mover.node, true);
+				console.log("Update model coordinates for ", mover);
+            	var scrollTop = document.getElementById("drawingPane").scrollTop;
+				var id = mover.node.id;
+				var index = 0
+				var initialString = "_" + _model.active.getInitialNodeString();
+				if(id.indexOf("_") > -1){
+					id = id.replace(initialString, "");
+					index = 1;
+				}
+				var node = registry.byId(mover.node);
+            	var widthLimit = document.documentElement.clientWidth - 110; 
+            	var topLimit = 20;
+
+            	if(g.x > widthLimit) {
+                	g.x = widthLimit;
+                	node.style.left = widthLimit+"px";
+            	}    
+
+            	if(g.x < 0) { 
+                	g.x = 0; 
+                	node.style.left = "0px";
+            	}    
+
+            	if((g.y + scrollTop) < topLimit){
+                	//check if bounds inside
+                	if(g.y < topLimit) { // BUG: this g.y should be absolute coordinates instead
+                    	g.y = topLimit;
+                    	node.style.top = topLimit+"px";  // BUG: This needs to correct for scroll
+                	}
+            	}
+            	_model.active.setPosition(id, index, {"x": g.x, "y": g.y+scrollTop});
+			}, true);
```

Comment:
I think I skrewed up the indentation here. Would it be possible to open it in sublime and just convert indentation to tabs. 4 indentations to 1 tab space?